### ISSUE TITLE
The llvm test fails if static llvm libraries are not provided

### ISSUE THREAD

**Issue by williamh** [mesonbuild/meson#4360]
On Gentoo Linux at least, and maybe other distros, the static llvm libraries are not provided by default.
This means that the test suite fails because it is unable to statically link llvm.
The issue is in the following file:

test cases/frameworks/15 llvm/meson.build

In this file, the foreach loop  currently reads:

```
foreach static : [true, false]
```

We patch it like this to get passed the failure:

```
foreach static : [false]
```

I'm thinking we should dynamically generate this list based on what is on the system and use that to control the loop instead of hard coding the list.

Any thoughts?

@floppym

**Comment by nirbheek** [comment]
CCing @dcbaker

**Comment by dcbaker** [comment]
Yes, that seems to be the case still. Sigh, I thought I fixed this with the last patches.

**Comment by dcbaker** [comment]
@williamh can you test this and make sure it works for you? I've tested this with a build of LLVM that has -DLLVM_BUILD_SHARED_LIBS set to true like Gentoo does (which, btw, upstream LLVM recommends you *don't* do that),  as well as the more standard -DLLVM_BUILD_LLVM_DYLIB as well as with and without -DLLVM_LINK_LLVM_DYLIB.

**Comment by floppym** [comment]
@dcbaker I can confirm that this resolves the issue on Gentoo.
